test/unit/test-amp-story-player.js

```diff
@@ -138,4 +158,36 @@ describes.realWin('AmpStoryPlayer', {amp: false}, env => {
       expect(stories[3][IFRAME_IDX]).to.eql(undefined);
     }
   );
+
+  it('should register handlers at build time', async () => {
+    buildStoryPlayer();
+    const player = new AmpStoryPlayer(win, playerEl);
+
+    env.sandbox
+      .stub(player, 'initializeHandshake_')
+      .resolves(fakeMessagingPromise);
+
+    messagingMock.expects('registerHandler').withArgs('selectDocument');
+    messagingMock.expects('setDefaultHandler');
+
+    await player.buildCallback();
+  });
+
+  it('should navigate to next story when the last page of a story is tapped', async () => {
+    buildStoryPlayer();
+    const player = new AmpStoryPlayer(win, playerEl);
+    const nextSpy = env.sandbox.spy(player, 'next_');
+
+    env.sandbox
+      .stub(player, 'initializeHandshake_')
+      .resolves(fakeMessagingPromise);
+
+    await player.buildCallback();
+    player.layoutCallback();
+
+    const fakeData = {next: true};
+    fireHandler('selectDocument', fakeData);
+
+    expect(nextSpy).to.be.called;
```

Comment:
Let's assert that the next story is actually selected instead? From your previous tests it looks straightforward, and would give a much better coverage